server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -525,6 +533,8 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
         mergeSchedulerConfig = new MergeSchedulerConfig(this);
         gcDeletesInMillis = scopedSettings.get(INDEX_GC_DELETES_SETTING).getMillis();
         softDeleteEnabled = version.onOrAfter(LegacyESVersion.V_6_5_0) && scopedSettings.get(INDEX_SOFT_DELETES_SETTING);
+        translogPruningByRetentionLease = version.onOrAfter(LegacyESVersion.V_6_5_0) &&
+            scopedSettings.get(INDEX_TRANSLOG_RETENTION_LEASE_PRUNING_ENABLED_SETTING);
```

Comment:
Is this version by choice? How are other versions supposed to get this

Reply:
As we are leveraging existing retention lease infrastructure, we are going with this supported version.